src/app/metadata/features/dataset-expansion-panel/dataset-expansion-panel.component.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Component for the individual expansion panels for the metadata browser
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Component, inject, Input, signal } from '@angular/core';
+import { MatExpansionModule } from '@angular/material/expansion';
+import { DatasetSummary } from '@app/metadata/models/dataset-summary';
+import { Hit } from '@app/metadata/models/search-results';
+
+/**
+ * Component for the expansion panel for each dataset found in the search results
+ */
+@Component({
+  selector: 'app-dataset-expansion-panel',
+  imports: [MatExpansionModule],
+  templateUrl: './dataset-expansion-panel.component.html',
+  styleUrl: './dataset-expansion-panel.component.scss',
+})
+export class DatasetExpansionPanelComponent {
+  @Input()
+  hit!: Hit;
+
+  #http = inject(HttpClient);
+  summary = signal<DatasetSummary>({
+    title: '',
+    accession: '',
+    description: '',
+    type: [],
+    studies_summary: { count: 0, stats: { accession: '', title: '' } },
+    files_summary: { count: 0, stats: { format: [] } },
+    samples_summary: {
+      count: 0,
+      stats: { sex: [], tissues: [], phenotypic_features: [] },
+    },
+    experiments_summary: { count: 0, stats: { experiment_methods: [] } },
+  });
+
+  constructor() {}
+
+  /**
+   * Function to obtain the dataset summary data
+   */
+  GetContent() {
+    this.#http
+      .get(
```

Comment:
Move this to a MetadataService and use it here.

The MetadataService should of course go into app/metadata/services.

This gives us a nice separation of concerns. The component doesn't need to know anything about how to acquire the data via http.

Reply:
Done 👍